include/llbuild/BuildSystem/BuildExecutionQueue.h

```diff
@@ -114,13 +114,16 @@ class BuildExecutionQueue {
   virtual bool
   executeProcess(QueueJobContext* context,
                  ArrayRef<StringRef> commandLine,
-                 ArrayRef<std::pair<StringRef, StringRef>> environment) = 0;
+                 ArrayRef<std::pair<StringRef, StringRef>> environment,
+                 std::function<void(StringRef)> partialOutputCallback = nullptr) = 0;
 
```

Comment:
Another thing we need to handle here is splitting of the stdout/stderr streams; currently they are merged, but with this change we should probably split them and supply the output for each independently to the client. This will be annoying because we will need to move to using `select()` (or something like it) to pull the data off the output pipes. However, it is important we do this so that `stderr` output from the compiler isn't lost.